sisyphus/simple_linux_utility_for_resource_management_engine.py

```diff
@@ -426,7 +426,7 @@ def init_worker(self, task):
         engine_logpath = (
             os.path.dirname(logpath)
             + "/engine/"
-            + os.getenv("SLURM_JOB_NAME")
+            + os.getenv("SLURM_JOB_NAME", self.process_task_name(task.task_name()))
             + "."
             + job_id
             + "."
```

Comment:
would you also update the default of SLURM_ARRAY_TASK_ID to get the actual task number instead of a dummy 1?

Reply:
How do I get that?

The `array_task_id` here in this function is also via the env var. Maybe I could use `array_task_id` instead to not have the logic duplicated, but it's just the same.

I think the env var is the only source we can know the array task id here, or not?

Reply:
ok, if the symlink is anyway unused / broken, we don't need to change this.